packages/compiler-cli/src/ngtsc/partial_evaluator/test/evaluator_spec.ts

```
@@ -37,6 +37,9 @@ describe('ngtsc metadata', () => {
   it('map access works',
      () => { expect(evaluate('const obj = {a: "test"};', 'obj.a')).toEqual('test'); });
 
+  it('resolves undefined property access',
```

Hi Matt,

Thanks for contributing a fix!

Quick question: does it make sense to add one more test case with extra nesting level, like this:

```
evaluate('const obj: any = {}', 'obj.bar.baz')
```

I guess returning `undefined` makes sense with `obj.bar` expression, but I'm wondering if we still need to throw an error while evaluating `obj.bar.baz`?

Thank you.

cc @JoostK @alxhub

---

@AndrewKushnir we should probably add a test for that, although its outcome is not affected by this PR. When evaluating the property access `baz`, its LHS will evaluate to `undefined` and ngtsc's `accessHelper` will return a `DynamicValue`:

https://github.com/angular/angular/blob/f3444b8156b77a7690f335ee9d5ca3ebc1afa3e7/packages/compiler-cli/src/ngtsc/partial_evaluator/src/interpreter.ts#L373

We would end up in the same situation if the test code was `const obj: any = { bar: undefined }`, hence this PR does not affect that particular scenario.

In general, I do think we should update the static evaluator to *never* throw an error, as doing so prevents from being able to accurately tell where the problem is. We may want to introduce a `DynamicValue` that represents an error condition instead, so that we can properly report errors.

---

Thanks for additional info @JoostK, that makes sense. I don't want to delay this PR, so I think we can add extra tests later if needed. Thank you.